pkg/controller/admin/mobile_apps.go

```diff
@@ -33,11 +38,30 @@ func (c *Controller) HandleMobileAppsShow() http.Handler {
 			return
 		}
 
-		c.renderShowMobileApps(ctx, w, apps)
+		realms, err := c.db.GetRealms()
```

Comment:
I'm not a fan of this mutation happening in the controller layer (I think it belongs in the database layer). It might be worth timeboxing an hour or so with gorm to see if we can do this in the query somehow.